src/keyboard/mod.rs

```diff
@@ -0,0 +1,258 @@
+//! Utilities for keyboard handling
+//!
+//! This module provides utilities for keyboardand keymap handling: keymap interpretation
+//! and forwarding keystrokes to clients using xkbcommon.
+//!
+//! You can first create a `KbdHandle` using the `create_keyboard_handler` function in this module.
+//! The handle you obtained can be cloned to access this keyboard state from different places. It is
+//! expected that such a context is created for each keyboard the compositor has access to.
```

Comment:
I see. When I did finish the libinput bindings, you will see that we maybe need to change some stuff about the `InputBackend` anyway, as libinput is in fact tracking single devices much better then seats (which is what we currently want to do in the `InputBackend`).

I guess we will figuring that out then.